### PR TITLE
Upgrade app devcontainer base to newer tooling

### PR DESCRIPTION
**Pull request by inf17101**
The PR upgrades the app devcontainer base to `Ubuntu-24.04` image including fixes for podman. It also includes the starship shell for non-root users and a tmux setup.

The changes took over parts from the Ankaios devcontainer base (`.devcontainer/Dockerfile.base`), but merged it with the portable configuration of the devcontainer to also run containers in Docker Desktop when developing on MacOS.

The readme has also been updated.

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [x] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

File: tools/app_dev_container_base/containers.conf [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+ipcns="shareable"
+utsns="private"
+cgroupns="private"
+cgroups="enabled"
 log_driver = "k8s-file"
 [engine]
 cgroup_manager = "cgroupfs"
 events_logger="file"
-runtime="crun"
+runtime="crun"
```
Comment (krucod3) [inline tools/app_dev_container_base/containers.conf:12]:
```suggestion
runtime="crun"

```

**Review by krucod3 — COMMENTED** [review]
Small findings documented.

**Comment by inf17101** [comment]
I have tested the following on wsl2 ubuntu-22.04 and macOS silicon Docker Desktop:

```shell

## host container tests
docker run --privileged -it --rm -v $(pwd):/mnt/cwd app-ankaios-dev:test sh -c 'echo "Hello World" > /mnt/cwd/test.txt && cat /mnt/cwd/test.txt'
docker run --privileged -it --rm --user ankaios -v $(pwd):/home/ankaios/cwd app-ankaios-dev:test sh -c 'echo "Hello World" >> /home/ankaios/cwd/test2.txt && cat /home/ankaios/cwd/test2.txt'
## container-in-container tests

# networking
podman run -d --rm -p 8080:80 docker.io/nginx:latest
curl localhost:8080
podman run -d --rm --network host docker.io/nginx:latest
curl localhost
podman network create test_net
podman run -d --rm --network test_net --name web_server docker.io/nginx:latest
podman run -it --rm --network test_net docker.io/alpine:latest sh -c 'wget web_server && cat index.html'
# bind mount
mkdir ~/test/
echo "HelloWorld" > ~/test/test.txt
podman run -it --rm -v ~/test:/mnt/mount docker.io/alpine:latest sh -c 'echo "Second line" >> /mnt/mount/test.txt && cat /mnt/mount/test.txt'
```

It has worked.

File: tools/app_dev_container_base/containers.conf (hunk 1 above)
Comment (inf17101) [reply tools/app_dev_container_base/containers.conf]: I have added a space

**Review by krucod3 — APPROVED** [review]
LGTM 👍

**Pull request merged**